plugin_tests/widgetSpec.js

```diff
@@ -569,6 +597,82 @@ describe('control widget view', function () {
         expect(w.model.get('path')).toEqual([]);
     });
 
+    it('file', function (done) {
+        var arg, file = new girder.models.FileModel({id: 'model id', name: 'd'});
+
+        hProto.initialize = function (_arg) {
+            arg = _arg;
+            this.breadcrumbs = [];
+        };
+        hProto.render = function () {};
+
+        var w = new slicer.views.ControlWidget({
+            parentView: parentView,
+            el: $el.get(0),
+            model: new slicer.models.WidgetModel({
+                type: 'file',
+                title: 'Title',
+                id: 'file-widget'
+            })
+        });
+
+        w.render();
+        checkWidgetCommon(w);
+
+        girder.rest.mockRestRequest(function () {
+            return $.Deferred().resolve(JSON.stringify(file)).promise();
+        });
+        hProto.on('g:saved', function () {
+            console.log('here');
+            expect(w.model.value().name()).toBe('d');
+
+            expect(w.model.get('path')).toEqual([]);
+            girder.rest.unmockRestRequest();
+            done();
+        });
+        w.$('.s-select-file-button').click();
+        expect(arg.parentModel).toBe(girder.auth.getCurrentUser());
+        arg.onItemClick(file);
+    });
+
+    it('image', function (done) {
+        var arg, file = new girder.models.FileModel({id: 'model id', name: 'e'});
+
+        hProto.initialize = function (_arg) {
+            arg = _arg;
+            this.breadcrumbs = [];
+        };
+        hProto.render = function () {};
+
+        var w = new slicer.views.ControlWidget({
+            parentView: parentView,
+            el: $el.get(0),
+            model: new slicer.models.WidgetModel({
+                type: 'image',
+                title: 'Title',
+                id: 'image-widget'
+            })
+        });
+
+        w.render();
+        checkWidgetCommon(w);
+
+        girder.rest.mockRestRequest(function () {
+            return $.Deferred().resolve(JSON.stringify(file)).promise();
+        });
+        hProto.on('g:saved', function () {
+            console.log('here');
+            expect(w.model.value().name()).toBe('e');
+
+            expect(w.model.get('path')).toEqual([]);
+            girder.rest.unmockRestRequest();
+            done();
```

Comment:
It doesn't.  I was just delusional in thinking that the tests did what they were supposed to.